COORDINATE_SYSTEM2/COORDINATE_SYSTEM2/CoordinateSystem.cs

```diff
@@ -6,48 +6,123 @@
+    /// Містить MaxDistance() для пошуку максимальної відстані між точками.
+    /// Містить PrintAllDistances(), який друкує всі наявні відстані з Ліста.
+    /// </summary>
     internal class CoordinateSystem
     {
-        List<Point> listOfPoints = new List<Point>();
+        List<IFigure> listOfFigures = new List<IFigure>();
 
-        List<PointsDistance> pointsDistances = new List<PointsDistance>();
+        List<PointsDistance> listOfPointsDistances = new List<PointsDistance>();
 
-        public void Add(Point point)
+        public void Add(IFigure figure)
         {
-            foreach (Point point1 in listOfPoints)
+            listOfFigures.Add(figure);
+        }
+
+        public void PrintAreas()
+        {
+            foreach (IFigure figure in listOfFigures)
             {
-                PointsDistance distance = new PointsDistance(point, point1);
-                pointsDistances.Add(distance);
+                var area = figure.Area();
+                var pishovNahyi = area == 0 ? "N / A" : area.ToString();
+                Console.WriteLine(figure.ToString());
+                Console.WriteLine($"Area is {pishovNahyi}\n");
             }
+        }
 
-            listOfPoints.Add(point);
+        public void PrintPerimeter()
+        {
+            foreach (IFigure figure in listOfFigures)
+            {
+                var perimeter = figure.Perimeter();
+                var pishovNahyi = perimeter == 0 ? "N / A" : perimeter.ToString();
+                Console.WriteLine(figure.ToString());
+                Console.WriteLine($"Perimeter is {pishovNahyi}\n");
+            }
         }
 
         public void DisplayAllPoints()
         {
+            List<Point> listOfPoints = AddAllFiguresPointsIntoList();
+
             for (int i = 0; i < listOfPoints.Count; i++)
             {
                 Console.WriteLine($"point {i} : {listOfPoints[i]};");
             }
         }
+        
+        public List<Point> AddAllFiguresPointsIntoList()
+        {
+            List<Point> listOfPoints = new List<Point>();
+
+            foreach (IFigure innerFigure in listOfFigures)
+            {
+                foreach (Point point in innerFigure.GetPoints())
+                {
+                    listOfPoints.Add(point);
+                }
+            }
+
+            return listOfPoints;
+        }
+
+        /// <summary>
+        /// метод, що рахує і друкує всі відстані. 
+        /// Створюю два ліста, які набирають значень всіх існуючих точок.
+        /// перший цикл бере кожен елемент з першого ліста.
+        /// другий цикл йде по другому лісті, беручи кожен елемент ОКРІМ першого для пошуку відстані.
+        /// Після знаходження відстані - добавляю цб відстань в ліст відстаней та
+        /// КОЛИ другий ліст дійшов до свого кінця - видаляю перший елемент з ДРУГОГО ліста
+        /// аби уникнути повторень.
+        /// </summary>
+        public void PrintAllDistances()
```

Comment:
what would happen if you call this method twice?))

Reply:
Spoiler it will print 2xtimes distances, if you call it another time, you'll get 3xtimes distances and so on